core/extension/gdextension_manager.h

```diff
@@ -59,6 +60,9 @@ class GDExtensionManager : public Object {
 	Vector<String> get_loaded_extensions() const;
 	Ref<GDExtension> get_extension(const String &p_path);
 
+	bool class_has_icon_path(const String &p_class) const;
+	String class_get_icon_path(const String &p_class) const;
```

Comment:
```suggestion
	const String &class_get_icon_path(const String &p_class) const;
```
I think this should be possible.

Reply:
Didn't work in the end, because of `return ""` creating a local/temporary variable, and I'm not confident how to solve this.